### PR TITLE
Add the Prow job to build the prowgen tool into a Docker image

### PR DESCRIPTION
**Pull request by chizhg**
/cc @chaodaiG @cjwagner

### CODE REVIEW

File: tools/prowgen/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+WORKDIR /go/src/istio
+
+COPY . /go/src/istio
+
+RUN go install /go/src/istio/tools/prowgen/cmd/prowgen
+
+RUN go install github.com/mikefarah/yq/v4@v4.16.2
+
+FROM gcr.io/distroless/base:22bd467b41e5e656e31db347265fae118db166d9
```
Comment (howardjohn) [inline tools/prowgen/Dockerfile]: nit: use `static` instead of `base`

Comment (chizhg) [reply tools/prowgen/Dockerfile]: Done.
That was copied from another Dockerfile, so I made the change together.

**Comment by chaodaiG** [comment]
don't think I have enough context for reviewing this, @chizhg , do you just want an approval from us?

**Comment by chizhg** [comment]
> don't think I have enough context for reviewing this, @chizhg , do you just want an approval from us?

This is part of the work proposed in go/ng3-ci-configgen for generalizing the Prow config generation tooling.

**Review by chaodaiG — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Pull request merged**